### ISSUE TITLE
Replace swagger-all by a bundle holding only the necessary dependencies

### ISSUE THREAD

**Issue by kaikreuzer** [openhab/openhab2-addons#549]
When starting the Karaf distribution, I now get 3 lines from SLF4J in the console:

```
-------------------------------------------------------------------------
Welcome to openHAB 2.0.0-SNAPSHOT
-------------------------------------------------------------------------

Hit '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
Hit '<ctrl-d>' or type 'system:shutdown' or 'logout' to shutdown openHAB.

openhab> SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

Does anyone know what introduced this and how we can get rid off it again?
##

**Comment by maggu2810** [comment]
Please check the certificate creation bundle...

**Comment by maggu2810** [comment]
No, it is not the certificate creation bundle.

**Comment by maggu2810** [comment]
It is gone after I removed the "swagger-all" bundle.
So perhaps the bundle themselves or a dependent service / bundle.

**Comment by maggu2810** [comment]
All the time it seems to disappear it appears again...

**Comment by maggu2810** [comment]
Could you try a
* <code>bundle:stop "Swagger Provider" "swagger-all"</code>
and restart your instance?

Then wait a moment after you realize it does not occur.

Now type
* <code>bundle:start "Swagger Provider"</code>

**Comment by kaikreuzer** [comment]
ok, it seems you have found the culprit!

**Comment by kaikreuzer** [comment]
I guess the problem is that the swagger-all bundles slf4j, these classes are started, but do not have any configuration.
So to solve this, we should really remove swagger-all and add a bundle with only the really required dependencies.

**Comment by maggu2810** [comment]
At least the bundled slf4j is solved using recent shk.

**Comment by kaikreuzer** [comment]
How?

**Comment by maggu2810** [comment]
The same way as I fixed the missing import in swagger-all before.
Checkout, fix, set a sp version, build, deploy, use it in the feature.
And last but not least, make a pull request.

**Comment by kaikreuzer** [comment]
You're the best :-)
But why didn't you take the chance to throw out even more ballast from that jar? (thinking about apache-commons or joda, which are both quite fat)

**Comment by maggu2810** [comment]
But why didn't you take the chance to throw out even more ballast from that jar?

Just a matter of time.

**Comment by wborn** [comment]
Is this still an open issue that should be resolved in some other repository? The slf4j warnings no longer show and it doesn't look like the Maven repo containing the bugfix bundles is still being used.

**Comment by maggu2810** [comment]
@wborn I fixed it long time ago: eclipse/smarthome@da988a7ecc9e8663e327012f43e4f0a71322507c

**Comment by wborn** [comment]
Thanks providing the evidence that this issue can be closed @maggu2810 ! :+1:
